### PR TITLE
Export iterator for SetNodeMap

### PR DESCRIPTION
**Pull request by jennybuckley**
Create iterator over the keys of SetNodeMap, mirroring https://github.com/kubernetes-sigs/structured-merge-diff/blob/master/fieldpath/element.go#L179-L184 for the elements of PathElementSet

### CODE REVIEW

**Comment by apelisse** [comment]
/lgtm
This seems correct.

/hold
is that the right thing to do @lavalamp ?

**Comment by jennybuckley** [comment]
If this works then https://github.com/jennybuckley/kubernetes/commit/5b0bc2f1cc1e7391cd532d3e3e6f6bfcc0c04c66 will be to add the conversion

**Comment by lavalamp** [comment]
It's either this or construct & return a list, I think I slightly prefer a list but not to the extent that I'd ask for a change.

Either way a test would be nice :)

**Comment by jennybuckley** [comment]
Okay I added a test for this iterator and also the existing one

File: fieldpath/element_test.go [hunk 1]
```diff
@@ -29,3 +29,20 @@ func TestPathElementSet(t *testing.T) {
...
+func TestPathElementSetIterate(t *testing.T) {
+	set := &PathElementSet{}
+	toAdd := 5
+	for i := 0; i < toAdd; i++ {
+		p := i
+		pe := PathElement{Index: &p}
+		set.Insert(pe)
+	}
+	count := 0
+	set.Iterate(func(_ PathElement) {
```
Comment (lavalamp) [inline fieldpath/element_test.go]: (It doesn't hurt to add this but I think this was effectively tested by the test for `String()`?)

File: fieldpath/element_test.go [hunk 2]
```diff
@@ -29,3 +29,20 @@ func TestPathElementSet(t *testing.T) {
...
+		count++
```
Comment (lavalamp) [inline fieldpath/element_test.go]: optional nit: I'd verify that you get the right items back, too, personally

**Comment by lavalamp** [comment]
/lgtm

meant to do that earlier sorry

**Pull request merged**